libdeno/libdeno_test.js

```diff
@@ -146,3 +146,50 @@ global.Shared = () => {
   ui8[1] = 43;
   ui8[2] = 44;
 };
+
+global.LibDenoEval = () => {
+  const [result, errInfo] = libdeno.eval("let a = 1; a");
+  assert(result === 1);
+  assert(!errInfo);
+  const [result2, errInfo2] = libdeno.eval("a = a + 1; a");
+  assert(result2 === 2);
+  assert(!errInfo2);
+};
+
+global.LibDenoEvalError = () => {
+  const [result, errInfo] = libdeno.eval("not_a_variable");
+  assert(!result);
+  assert(!!errInfo);
+  assert(errInfo.isNativeError); // is a native error (ReferenceError)
+  assert(!errInfo.isCompileError); // is NOT a compilation error
+  assert(errInfo.thrown.message === "ReferenceError: not_a_variable is not defined");
+
+  const [result2, errInfo2] = libdeno.eval("throw 1");
+  assert(!result2);
+  assert(!!errInfo2);
+  assert(!errInfo2.isNativeError); // is NOT a native error
+  assert(!errInfo2.isCompileError); // is NOT a compilation error
+  assert(errInfo2.thrown === 1);
```

Comment:
I guess this is all about getting more detailed stack traces? I'm not sure it makes sense to expose this when `Error.stack` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack)  exists... Generally we should be very careful about adding new libdeno bindings.

Reply:
`thrown` is just the actual thing that is getting thrown. `isNativeError` is used to decide whether the thrown should be printed and formatted as an Error (I remember there were some caveats of using `instanceof Error` here) (also, I don’t think that printing stack for anything that has a 'stack' property sound like a good idea...), while `isCompileError` is used to decide whether an error happens in compile time of the provided script, allowing Deno to distinguish `{` and `eval('{'), and is used in #1420